### PR TITLE
Solves issue #278

### LINKED ISSUE #278: Recovering a partially created order
## Expected behaviour

When an order is not fully created for any reason and the valid order is not sent back to Shopping Flux Web Service, then we should be able to complete the order at the next getOrders batch.

## Actual behaviour

The failed order is not completed.

## Steps to reproduce the behaviour

1 - getOrders
2 - Having a third party buggy module activated that fails the order creation process (will not send the validOrders)
3 - Getting the orders again

### PR DESCRIPTION
**Pull request by Vincent-Profileo**
# Fixes #278

## Changes made
Added mecanism to detect previous order partially created, re set prices and carrier, and notify ShoppingFlux

### CODE REVIEW

File: shoppingfluxexport.php [hunk 1]
```diff
@@ -1577,6 +1578,37 @@ public function hookbackOfficeTop($no_cron = true, $forcedOrder = false)
...
+                                AND c.email = 'do-not-send@alerts-shopping-flux.com'
+                                AND (c.lastname = '" . (string)$order->BillingAddress->LastName . "' OR c.firstname = '" . (string)$order->BillingAddress->LastName . "')
+                                AND a.id_address = o.id_address_delivery
+                                AND a.alias LIKE '%" . $orderExists['id_order'] . "%'
+                                ORDER BY o.id_order DESC
+                            ");
+                            
+                            if (! $forcedOrder && isset($orderExists['id_order']) && isset($orderExists['id_order'])) {
+                                // This is the second try of an order creation, last process could not be completed
+                                SfLogger::getInstance()->log(SF_LOG_ORDERS, 'Order allready exists (id = ' . $order->IdOrder . '): notifying ShoppingFlux', $doEchoLog);
```
Comment (clement-hollander) [inline shoppingfluxexport.php]: already

Comment (clement-hollander) [reply shoppingfluxexport.php]: @Vincent-Coll tu peux modifier allready par already ? merci

**Comment by Vincent-Profileo** [comment]
@clement-hollander 
my bad, corrected on 
https://github.com/shoppingflux/shoppingfluxexport/pull/299/commits/a199bce9fc61c2604d8fb58ccb5012af8b864dd7

File: shoppingfluxexport.php [hunk 2]
```diff
@@ -1569,14 +1569,46 @@ public function hookbackOfficeTop($no_cron = true, $forcedOrder = false)
...
+                            $orderExists = Db::getInstance()->getRow("SELECT * FROM " . _DB_PREFIX_ . "orders o, " . _DB_PREFIX_ . "customer c, " . _DB_PREFIX_ . "address a
+                                WHERE o.id_customer = c.id_customer
+                                AND c.email = 'do-not-send@alerts-shopping-flux.com'
+                                AND (c.lastname = '" . (string)$order->BillingAddress->LastName . "' OR c.firstname = '" . (string)$order->BillingAddress->LastName . "')
+                                AND a.id_address = o.id_address_delivery
+                                AND a.alias LIKE '%" . $orderExists['id_order'] . "%'
+                                ORDER BY o.id_order DESC
+                            ");
+                            
+                            if (! $forcedOrder && isset($orderExists['id_order']) && isset($orderExists['id_order'])) {
```
Comment (clement-hollander) [inline shoppingfluxexport.php]: y'a deux fois la meme conditions la

**Review by clement-hollander — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
@clement-hollander any other changes required on this ?

**Pull request merged**